src/pages/viewProfile/index.js

```diff
@@ -0,0 +1,176 @@
+import { useState, useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+import Button from "../../components/button";
+import "../../pages/viewProfile/viewProfile.css";
+import Card from "../../components/card";
+import ProfileCircle from "../../components/profileCircle";
+import { useParams } from "react-router-dom";
+import { get } from "../../service/apiClient";
+import jwt_decode from "jwt-decode";
+import ErrorMessage from "../../components/errorMessage";
+
+const initialState = {
+  id: "",
+  email: "",
+  role: "",
+  cohort_id: "",
+  firstName: "",
+  lastName: "",
+  biography: "",
+  githubUrl: "",
+  specialism: "",
+};
+// TODO:
+// update the hard-coded start date, end date and job title in the initial state and update it on jsx too. (when server team updates data)
+
+function ViewProfile() {
+  // STATES
+  const [profile, setProfile] = useState(initialState);
+  const [loggedInUserInfo, setLoggedInUserInfo] = useState({});
+  const [isError, setIsError] = useState(false);
+
+  // GLOBAL VARIABLES
+  const navigate = useNavigate();
+  const { id } = useParams();
+  const token = localStorage.getItem("token");
+
+  // Checks which user is logged in
+  const getUserInfo = async () => {
+    const { userId } = jwt_decode(token);
+    const res = await get(`users/${userId}`);
+    setLoggedInUserInfo(res.data.user);
+  };
```

Comment:
There are going to be multiple places where getting the current logged in users data will be used. I recommend this kind of functionality should be moved to the AuthProvider in `src/context/auth.js`. Review the code in there to see how it works.

The token is set in there when a user logs in, registers & completes the create profile step, or loads the app for the first time and a token is loaded from localStorage. You can add some functionality to not only save the token in the context state but also save the user's data in context state